internal/incident/incident.go

```diff
@@ -320,61 +422,119 @@ func (i *Incident) evaluateEscalations(ctx context.Context, tx *sqlx.Tx, ev *eve
 				if escalation.Condition == nil {
 					matched = true
 				} else {
-					cond := &rule.EscalationFilter{
-						IncidentAge:      time.Now().Sub(i.StartedAt),
-						IncidentSeverity: i.Severity,
+					// evaluateCond evaluates the specified escalation filter 1 nanosecond before, at and 1 nanosecond
+					// after the configured IncidentAge filter and returns error on any evaluation failure.
+					// This anonymous function serves simply for preventing duplicate codes.
+					evaluateCond := func(cond *rule.EscalationFilter) (bool, error) {
+						before := &rule.EscalationFilter{
+							IncidentAge:      cond.IncidentAge - time.Nanosecond,
+							IncidentSeverity: cond.IncidentSeverity,
+						}
+						// Evaluate whether the escalation filter evaluates to true 1 nanosecond
+						// before the configured incident age condition.
+						matched, err := escalation.Condition.Eval(before)
+						if err != nil {
+							return false, err
+						}
+
+						if !matched {
+							// Evaluate the configured escalation filter without any modifications.
+							matched, err = escalation.Condition.Eval(cond)
+							if err != nil {
+								return false, err
+							}
+						}
+
+						if !matched {
+							// Evaluate whether the escalation filter evaluates to true 1 nanosecond
+							// after the configured incident age condition.
+							return escalation.Condition.Eval(
+								&rule.EscalationFilter{IncidentAge: cond.IncidentAge + time.Nanosecond, IncidentSeverity: cond.IncidentSeverity},
+							)
+						}
+
+						return matched, nil
```

Comment:
I know that I suggested evaluating the rule at incident_age±1ns to capture strange uses of the `=` and `>` operator there, but now it feels like we're just making the code here more complex when we should probably tell users, that they should just go with `incident_age>=x` and maybe some others like `incident_age<x`. But for sure a restricted set of operators, because for `incident_age>=1h` it's obvious what should happen. Do we really want to make sure that `incident_age=1h` and `incident_age>1h` behave the same, or just offer `>=` in the web module for `incident_age`?